tests/foreman/api/test_ansible.py

```diff
@@ -470,3 +470,72 @@ def test_positive_ansible_job_on_multiple_host(
         assert result.succeeded == 2  # SELECTED_ROLE working on rhel8/rhel9 clients
         assert result.failed == 1  # SELECTED_ROLE failing  on rhel7 client
         assert result.status_label == 'failed'
+
+    @pytest.mark.no_containers
+    @pytest.mark.rhel_ver_list('8')
+    def test_negative_ansible_job_timeout_to_kill(
+        self, target_sat, module_org, module_location, module_ak_with_synced_repo, rhel_contenthost
+    ):
+        """when running ansible-playbook, timeout to kill/execution_timeout_interval setting
+            is honored for registered host
+
+        :id: a082f599-fbf7-4779-aa18-5139e2bce777
+
+        :steps:
+            1. Register a content host with satellite
+            2. Run Ansible playbook to pause execution for a while, along with
+                timeout to kill/execution_timeout_interval setting on the registered host
+            3. Verify job is terminated honoring timeout to kill setting and verify job output.
+
+        :expectedresults: Timeout to kill terminates the job if playbook doesn't finish in time
+
+        :BZ: 1931489
+
+        :customerscenario: true
+        """
+        playbook = '''
+            ---
+            - name: Sample Ansible Playbook to pause
+              hosts: all
+              gather_facts: no
+              tasks:
+              - name: Pause for 5 minutes
+                pause:
+                  minutes: 5
+        '''
+        result = rhel_contenthost.register(
```

Comment:
Can we use api method to register here?

Reply:
there are other tests using it, I'd consider changing it in separate PR if required